indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexPhaseRunner.java

```diff
@@ -289,13 +289,15 @@ private void stopInternal()
   @Override
   public void collectReport(SubTaskReportType report)
   {
-    taskMonitor.collectReport(report);
+    if (taskMonitor != null) {
+      taskMonitor.collectReport(report);
+    }
```

Comment:
`collectReport` should be called only when there is a subtask sending its report. Since `TaskMonitor` is responsible for spawning subtasks, it seems quite strange if `taskMonitor` is null when this method is called. How about adding a precondition that explodes when it's null?

Reply:
I just saw that the `taskMonitor` was accessed without a null check so I added the null check just in case. The bug mentioned in this ticket can be fix with just the change in `getReports()`. Given the context from your comment, maybe it is better to leave it as is and adds a comment mentioning something like... collectReport is only called when there is a subtask sending its report. Since TaskMonitor is responsible for spawning subtasks, the taskMonitor cannot be null`

Reply:
The comment sounds good. I think having a null check is also a good convention because someone like me could break the contract around `taskMonitor` unintentionally in the future. Perhaps we can add `assert taskMonitor != null;`. This will be ran only in tests, but not in production.

Reply:
Is there a way to have static analysis flag all these uses? I've tried using combinations of `@Nullable`(doesn't realize that once it's set, it's good for the rest of the method) and `@MonotonicNonNull` but haven't been able to get intelliJ to flag these issues for me locally yet. I figured I would just leave a comment here and maybe you will have better luck

Reply:
You can set the severity of the "Nullability problems" intellij inspection to error instead of warning. I think we haven't done it because there could be lots of places where the inspection will complain about nulls. It could be worth doing it at some point though.

Reply:
Done.